### ISSUE TITLE
corOutCoverage clarification

### ISSUE THREAD

**Issue by igordot** [marbl/canu#230]
I am trying to understand the `corOutCoverage` parameter. From FAQ:

> The first step in Canu is to find high-error overlaps and generate corrected sequences for subsequent assembly. This is currently the fastest step in Canu. By default, only the longest 40X of data (based on the specified genome size) is used for correction. If you have a dataset with uneven coverage or small plasmids, correcting the longest 40X may not give you sufficient coverage of your genome/plasmid.

The best explanation I could find is from https://github.com/marbl/canu/issues/66 : 

> it takes the reads sorted from longest to shortest, keeps adding in reads until it reaches 40X, then uses the remaining shorter reads to correct those longest reads? So I need to increase corOutCoverage to make it correct more reads? Presumably there is a trade-off here, where going too high with corOutCoverage will reduce the accuracy of the corrected reads. 

I think means that the longest reads are processed until they are piled up to 40X and then the remaining reads are used for correction. If the coverage is perfectly even, the whole genome is covered at 40X. If the coverage is uneven, it will range from 0 to 40X, but correction happens for all reads that made it before coverage hit 40X somewhere, so even the long reads with no overlaps are corrected. Is that correct?

Does that mean that the corrected long reads are kept for assembly in the later steps and the short reads are discarded after they are used only for correction?

What happens if you set corOutCoverage to a very high number so it is never reached? Is there no correction then?

**Comment by skoren** [comment]
Yes, you are correct. The corOutCoverage is used to compute the bases to add up to 40*genomeSize. Given the overlaps for all reads, Canu will estimate the corrected read length and keep enough reads to get the requested correct bases. Since this is an estimate, you usually end up with 30-35X instead of 40. Setting corOutCoverage > your input coverage will correct all reads.

A long time ago the corOutCoverage was also used to limit overlaps (so only 40X of the longest reads were overlapped to all reads) but since MHAP has gotten faster, this is no longer done and all-vs-all overlaps are computed. The corOutCoverage serves as an optimization on the downstream trimming and unitigging steps so you're not computing on as much data.

**Comment by igordot** [comment]
It sounds like increasing corOutCoverage will only slow down the whole process. Based on previous discussion, I thought it would make the correction worse.

**Comment by skoren** [comment]
Yes, just a speed issue, shouldn't affect the quality of the corrected reads, they would all be independently corrected.

**Comment by igordot** [comment]
I repeated with higher corOutCoverage setting. Increasing the coverage slows the process by a lot (from hours to days), so I understand why it's relatively conservative by default.

I had a similar results as the previous issue I referenced (https://github.com/marbl/canu/issues/66) where a large contig ended up getting split into two with higher coverage cutoff. If higher coverage results in more reads being used, why does the split happen if the reads that were previously bridging the two regions are still there? It's probably a complicated situation, but maybe there is a simple explanation.
